src/main/java/com/aws/greengrass/clientdevices/auth/api/Result.java

```diff
@@ -27,28 +38,49 @@ public static <V> Result<V> ok() {
         return ok(null);
     }
 
-    public static <E extends Exception> Result<E> error(E value) {
-        return new Result<>(Status.ERROR, value);
+    public static <T, E extends Exception> Result<T> error(E err) {
```

Comment:
This changes to result are such that we apply generic more easily and do things like

```java
Result<CAConfiguration> someMethod() {
     if (badThing) {
          Result.error(new SomeException("blah");
     }

     return Result.ok(CAConfiguration);
}
```